### DISCUSSION TITLE
Editable TextDocumentContentProvider

### DISCUSSION THREAD

**Discussion by xpe** [microsoft/vscode-discussions#855] [Extension Development QnA] [answered]
I would like to write an extension that allows editing of a (virtual) concatenation of files in a directory.

Here is an example of a [virtual document] I'd like to edit:

```
=====[ file_1.markdown ]=====
# File 1
This is the first file.
=====[ file_2.markdown ]=====
# File 2
This is the second file.
=====[ file_3.markdown ]=====
# File 3
This is the third file.
```

It seems to me that a `TextDocumentContentProvider` is _almost_ what I want, but according to the documentation, it is read-only. Others have noticed this limitation and asked to make it read-write: see this GitHub thread that started in 2016:

- [Editable TextDocumentContentProvider for extensions][Issue_10548]

[Issue_10548]: https://github.com/microsoft/vscode/issues/10547

[virtual document]: https://code.visualstudio.com/api/extension-guides/virtual-documents

**Comment by gjsjohnmurray** [comment, marked as answer]
I suggest you look at `FileSystemProvider`.
